.github/workflows/ci.yaml

```diff
@@ -58,29 +50,15 @@ jobs:
         override: true
         components: rustfmt, rust-src
 
-    - name: Cache cargo directories
-      uses: actions/cache@v2
```

Comment:
Don't have strong opinion about whether we should hand-roll or use an action. I personally lean towards keeping as much logic as possible in Rust in the repo itself, but at this point I think @lnicola word about CI weighs more that mine :0)

However, if we do use an action not from the official actions repo, let's specify it todether with a sha, to protect from supply chain attacks and to ensure reproducability. 

Actions themselves don't have lockfiles, with all associated problems.

Reply:
Agreed about the hash.

My thought process here was that setting up the cache is a moderately complex thing which could be (re)used by a lot of projects. So it makes sense to delegate maintaining it to someone else :-).

Reply:
On the one hand, it is true. On the other hand, not messing this up is hard :-) Like, the Travis used to have `cache: cargo`, and that was broken. Or the "make a github relase" thing, for which we cribbed off some alexcrichton code from wasmtime because various re-usable actions don't work.

Another issue is re-suability -- by having cache  cleaning implemented in rust in the repo, the logic can be re-used for whatever CI system we use, while javascript actions tie us to actions ecosystem. 

Finally, there's an issue that re-usability should ideally be applied in the right place. tar issue should be fixed on MacOS / GH builders, cache issue should be fixed in Cargo, so that `target` has different directories for stable and volatile things.

But, again, this is all "If I were a king of the world, I'd rule thus" kind of thing, practicality beats purity most of the times!